lib/fastlane/actions/appium.rb

```diff
@@ -0,0 +1,145 @@
+module Fastlane
+  module Actions
+    class AppiumAction < Action
+      INVOKE_TIMEOUT = 30
+
+      def self.run(params)
+        require 'rspec'
+        require 'appium_lib'
+
+        Helper.log.info "Parameter Platform: #{params[:platform]}"
+        Helper.log.info "Parameter Spec Path: #{params[:spec_path]}"
+        Helper.log.info "Parameter App Path: #{params[:app_path]}"
+        Helper.log.info "Parameter Invoke Appium Server: #{params[:invoke_appium_server]}"
+        Helper.log.info "Parameter Appium Server Host: #{params[:host]}"
+        Helper.log.info "Parameter Appium Server Port: #{params[:port]}"
+        Helper.log.info "Parameter Appium Capabilities: #{params[:caps]}"
+
+        if params[:invoke_appium_server]
+          appium_pid = invoke_appium_server(params)
+          wait_for_appium_server(params)
+        end
+
+        configure_rspec(params)
+
+        rspec_args = []
+        rspec_args << params[:spec_path]
+        status = RSpec::Core::Runner.run(rspec_args).to_i
+        if status != 0
+          raise "Failed to run appium spec. status code: #{status}".red
+        end
+      ensure
+        Actions.sh "kill #{appium_pid}" if appium_pid
+      end
+
+      def self.invoke_appium_server(params)
+        if `which appium`.to_s.length == 0
+          raise 'You have to install appium using `npm install -g appium`'.red
+        end
+
+        fork do
+          Process.exec("appium -a #{params[:host]} -p #{params[:port]}")
+        end
+      end
+
+      def self.wait_for_appium_server(params)
+        loop.with_index do |_, count|
+          break if `lsof -i:#{params[:port]}`.to_s.length != 0
+
+          if count * 5 > INVOKE_TIMEOUT
+            raise 'Invoke appium server timed out'.red
+          end
+          sleep 5
+        end
+      end
+
+      def self.configure_rspec(params)
+        RSpec.configure do |c|
+          c.before(:each) do
+            caps = params[:caps] || {}
+            caps[:platformName] ||= params[:platform]
+            caps[:autoAcceptAlerts] ||= true
+            caps[:app] = params[:app_path]
+
+            @driver = Appium::Driver.new(
+              caps: caps,
+              server_url: params[:host],
+              port: params[:port]
+            ).start_driver
+            Appium.promote_appium_methods(RSpec::Core::ExampleGroup)
+          end
+
+          c.after(:each) do
+            @driver.quit unless @driver.nil?
+          end
+        end
+      end
+
+      def self.description
+        'Run integration spec by Appium'
+      end
+
+      def self.available_options
+        [
+          FastlaneCore::ConfigItem.new(
+            key: :platform,
+            env_name: 'FL_APPIUM_PLATFORM',
+            description: 'Appium platform name',
+            is_string: true
+          ),
+          FastlaneCore::ConfigItem.new(
+            key: :spec_path,
+            env_name: 'FL_APPIUM_SPEC_PATH',
+            description: 'Appium spec directory path',
```

Comment:
I don't use Appium myself, but aren't there some default values? If for example there is always a `spec` folder, you can add

```
Dir["./spec"].last
```

which will either be nil (if not there) or a path (if available)